module/VuFind/src/VuFind/Role/PermissionProvider/Cookie.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * Cookie permission provider for VuFind.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2007.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Authorization
+ * @author   Alex Buckley <alexbuckley@catalyst.net.nz>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     http://www.vufind.org  Main Page
+ */
+
+namespace VuFind\Role\PermissionProvider;
+
+use VuFind\Cookie\CookieManager;
+
+/**
+ * Cookie permission provider for VuFind.
+ *
+ * @category VuFind
+ * @package  Authorization
+ * @author   Alex Buckley <alexbuckley@catalyst.net.nz>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+class Cookie implements PermissionProviderInterface
+{
+    use \VuFind\Log\LoggerAwareTrait;
+
+    /**
+     * Constructor
+     *
+     * @param CookieManager $cookieManager Cookie manager
+     */
+    public function __construct(
+        protected CookieManager $cookieManager
+    ) {
+    }
+
+    /**
+     * Return an array of roles which may be granted the permission based on
+     * the options.
+     *
+     * @param mixed $options Options (cookieNames) provided from configuration.
+     *
+     * @return array
+     */
+    public function getPermissions($options)
+    {
+        foreach ((array)$options as $cookieName) {
+            $this->debug("getPermissions: option cookieName '{$cookieName}'");
+            $cookie = $this->cookieManager->get($cookieName);
```

Comment:
I wonder if we need some kind of mechanism for accounting for cookie values. If the cookie is just used as a flag to grant public access, then there's not really a security concern, but if we wanted a greater level of security, it might be better if we had some way of generating a random token in the session and then validating the cookie against that value, so people couldn't just edit their cookies to break in. That might be more work/complexity than is needed for your use case, and I certainly don't insist on over-engineering this... but if we keep it simple, we should include appropriate warnings in the documentation about any limitations related to ease of spoofing, etc.

Reply:
thanks @demiankatz I'm going to add these two remaining changes next!

Reply:
thanks @demiankatz we are thinking about if the simple solution is best for our use case or not. We'll get back to you shortly.